src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -140,6 +140,19 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource)(implicit ec: E
     }
   }
 
+  // Name & namespace for new methods are validated against the regex.
+  // Existing methods get a pass because cleaning up the repo is prohibitively difficult (GAWB-1614)
+  private def validateNamesForNewEntity[T](entity: AgoraEntity)(op: => ReadWriteAction[T]): ReadWriteAction[T] = {
+    val nameRegex = """[a-zA-Z0-9-_.]+""".r
```

Comment:
since this is a constant, create it somewhere outside this method so we don't need to create it every time. Maybe in a companion object to AgoraBusiness.